### ISSUE TITLE
Analysis report part 2 feedback

### ISSUE THREAD

**Issue by ugurdogrusoz** [boranyldrm/CS319-2c#4]
Looks like you only partially addressed the concerns in the first part of your report.
I am still worried that your project will have minimum functionality (overly simple).
Regarding the second part of the report:
- Object model could be better
- Dynamic model could be improved (more SDs are needed)
- UI model is missing the key mockup screenshot for ingame play
Detailed comments: 
[2C.analysis.2.pdf](https://github.com/boranyldrm/CS319-2c/files/852408/2C.analysis.2.pdf)

**Comment by BerkeSoysal** [comment]
Dear Instructor,
Should/Can we add extra features on gameplay that we didn't mention on our Analysis report, to make it a more complex project?

**Comment by ugurdogrusoz** [comment]
Yes you may add new features but you should revise your analysis report accordingly before design. Make sure the revised report is uploaded with an appropriate name.

**Comment by BerkeSoysal** [comment]
Dear Instructor,
While revising the analysis report, I'm trying to be sure that I'm not doing any mistakes, so could you please give me feedback about this particular sequence diagram?
[InitializeGameScenario.pdf](https://github.com/boranyldrm/CS319-2c/files/865212/InitializeGameScenario.pdf)
I'm specifically not sure about drawMap. Should we let the GameEngine draw the map or call DisplayManager? And for the rest of the game, should we call for DisplayManager for each time repainting or handle it on GameEngine?
Secondly, because we don't want to boundary objects to initialize objects, I created a Top class. But it looks silly on the class diagram. Is something wrong here?
![finishedclass](https://cloud.githubusercontent.com/assets/16117551/24266631/2284443e-1010-11e7-837b-e4b09312a158.png)

**Comment by ugurdogrusoz** [comment]
It is in much better shape but: 1) some of my comments in original document aren't addressed 2) there are still problems (see attached); your sequence diagram is still not very organized in the sense that I cannot clearly see boundary objects calling control ones, which in turn call entity objects.
Regarding the class diagram, just the fact that you named a class "Top" tells me that you don't know what it is supposed to do! Btw, initialization of main menu or game engine is not a big deal; it could be done by the main method. What's more important is when and by whom are maps, levels, entity objects are created/initialized.
[InitializeGameScenario.UD.pdf](https://github.com/boranyldrm/CS319-2c/files/867006/InitializeGameScenario.UD.pdf)
If you need further feedback, you need to stop by my office, communication over Github or email is not effective (at least for me).
